feat(eslint-plugin): add config that disables type-aware linting (#6470)
typescript-eslint/typescript-eslint: docs/linting/Configurations.mdx

### `base`

A minimal ruleset that sets only the required parser and plugin options needed to run typescript-eslint.
We don't recommend using this directly; instead, extend from an earlier recommended rule.

This config is automatically included if you use any of the recommended configurations.

See [`configs/base.ts`](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/configs/base.ts) for the exact contents of this config.

### `disable-type-checked`

A utility ruleset that will disable type-aware linting and all type-aware rules available in our project.

> This config makes it easy to have your base config concerned with type-aware linting, and then conditionally use [overrides](https://eslint.org/docs/latest/use/configure/configuration-files#configuration-based-on-glob-patterns) to disable type-aware linting on specific subsets of your codebase.

Reviewer: Nit: _"easy"_ as a term can be offputting to developers who aren't finding the concepts easy. Which is a lot of folks when it comes to linting configs & the like IME.

```suggestion
This config is useful if you'd like have your base config concerned with type-aware linting, and then conditionally use [overrides](https://eslint.org/docs/latest/use/configure/configuration-files#configuration-based-on-glob-patterns) to disable type-aware linting on specific subsets of your codebase.
```
